### PR TITLE
Folder Structure documentation improvements

### PR DESCRIPTION
**Pull request by brunobowden**
- Capitalize `Objective-C`
- remove `j2objcBuildAllLibraries` from task list
- correct j2objcCycleFinder task enable

Folder Structure
- Alphabetical ordering
- Indicate which are folders
- Fix j2objcCycleFinder enable docs
- .gitignore for excluded files
- Note which are generated files / folders

### CODE REVIEW

**Comment by brunobowden** [comment]
@advayDev1, @confile - this should be straight forward

File: README.md [hunk 1]
```diff
@@ -46,47 +47,52 @@ Within the Android application's project `build.gradle`, make it dependent on th
...
-    │   ├── lib                        // library dependencies, must have source code
-    │   │   └── lib-with-src.jar       // library with source is translated
-    │   └── src                        // src/main/java/... shared code for translation
-    └── ios
-        ├── IosApp.xcworkspace         // Xcode workspace
-        ├── IosApp.xcodeproj           // Xcode project, which is modified by j2objcXcode / CocoaPods
-        ├── IosApp                     // j2objcXcode configures dependency on j2objcOutputs/{libs|src}
-        ├── IosAppTests                // j2objcXcode configures the same except "debug" libraries
-        └── Podfile                    // j2objcXcode modifies this file for use by CocoaPods
+    │   ├── settings.gradle            // include ':android', ':shared'
```
Comment (confile) [inline README.md]: Dis is the wrong place for settings.gradle you have to put it one level above. The problem is that when your shared project includes subprojects you need that settings.gradle too.

File: README.md [hunk 2]
```diff
@@ -46,47 +47,52 @@ Within the Android application's project `build.gradle`, make it dependent on th
...
+    │   └── src/...                    // src/main/java/... only Android specific code
+    ├── ios
+    │   ├── iosApp.xcworkspace         // Xcode workspace
+    │   ├── iosApp.xcodeproj           // Xcode project, which is modified by j2objcXcode / CocoaPods
+    │   ├── iosApp/...                 // j2objcXcode configures dependency on j2objcOutputs/{libs|src}
+    │   ├── iosAppTests/...            // j2objcXcode configures as above but with "debug" buildType
+    │   ├── Podfile                    // j2objcXcode modifies this file for use by CocoaPods
+    │   └── Pods/...                   // CocoaPods managed folder for Xcode, not checked in
```
Comment (confile) [inline README.md]: I think we should omit auto generated folder in this description. The user does not know which folder he has to created and which not.

Comment (advayDev1) [inline README.md]: agreed.  @brunobowden, if you have a project working with settings.gradle not at the root i'm interested to know how (are you just overriding the subproject path with ../?)

File: README.md (hunk 1 above)
Comment (advayDev1) [inline README.md]: agreed. @brunobowden, if you have a project working with settings.gradle not at the root i'm interested to know how (are you just overriding the subproject path with ../?)

File: README.md (hunk 2 above)
Comment (advayDev1) [inline README.md]: Actually, I think the most important folders should be included.  You might just want to mark the ones that are not the user's responsibility with an asterisk or hashtag and a legend at the bottom though

**Comment by advayDev1** [comment]
LGTM with above comments

File: README.md (hunk 1 above)
Comment (confile) [inline README.md]: @brunobowden I am also interested in how this works. Must be magic.

File: README.md (hunk 2 above)
Comment (confile) [inline README.md]: @advayDev1 Agreed to @advayDev1 idea.

**Comment by brunobowden** [comment]
I've fixed the issues you reported with the folder structure. It was just wrong.

Also clarified which are generated folders and what should be listed in `.gitignore`

**Comment by advayDev1** [comment]
@brunobowden  - important: this task is gone in #189.  it never made much sense to use anyway.  j2objcAssemble is the only client-relevant task in building libraries

**Comment by brunobowden** [comment]
Removed `j2objcBuildAllLibraries` from task list, leaving behind `j2objcAssemble`

I think this should be good to go now.

**Comment by advayDev1** [comment]
LGTM (just Fix conflicts with #189 )

**Comment by advayDev1** [comment]
fyi: You may wish to change your commit's title as it doesn't exist anymore

**Comment by brunobowden** [comment]
Fixed conflicts, reworded title and commit... finally time to merge

**Pull request merged**